server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -351,6 +357,32 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point) thro
         return parseGeoPoint(parser, point, false);
     }
 
+    /**
+     * Parses the value as a geopoint. The following types of values are supported:
+     * <p>
+     * Object: has to contain either lat and lon or geohash fields
+     * <p>
+     * String: expected to be in "latitude, longitude" format or a geohash
+     * <p>
+     * Array: two or more elements, the first element is longitude, the second is latitude, the rest is ignored if ignoreZValue is true
+     */
+    public static GeoPoint parseGeoPoint(Object value, final boolean ignoreZValue) throws ElasticsearchParseException {
+        try {
+            XContentBuilder content = JsonXContent.contentBuilder();
+            content.value(value);
```

Comment:
Should we put the value under a field in a top-level object? My reasoning is that I wouldn't be surprised that we start failing producing malformed json documents at some point, which would be the case here if you give a top-level string which is not enclosed in an object?

Reply:
Yes, I was thinking about it as well. Why the current solution is a bit faster it indeed feels somewhat hacky. I will add a top level object.